odl/util/utility.py

```diff
@@ -163,6 +185,145 @@ def is_complex_floating_dtype(dtype):
     return np.issubsctype(dtype, np.complexfloating)
 
 
+def default_dtype(impl, field):
```

Comment:
Shouldnt this be possible to do in some smarter way, perhaps as a static function on `Ntuples` and `CudaNtuples`? That would be easier to extend.